### PR TITLE
Added working process and git documents

### PR DESCRIPTION
**Pull request by petertretyakov**
Trello task: To document working process
https://trello.com/c/Jdmgbh5s/11-to-document-working-process

- Added README
- Added Working Process Description
- Added Working with Git Description
- Added resources folder containing images and git hook for Trello integration (in scripts folder)
- Added standard macOS .gitignore (we all work from Macs, yes? :)

Read carefully. By merge you accept all that was written there.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,20 @@
- -
+# CocoaHeads Russia
+
+We are the biggest community of iOS/macOS Developers in Russia. We organize developers meetups in Moscow and St.Petersburg.
```
Comment (smejl) [inline README.md]: Saint-Petersburg is better)

File: README.md [hunk 2]
```diff
@@ -1 +1,20 @@
...
+
+Follow us on [Twitter](https://twitter.com/cocoaheadsru) and [Facebook](https://www.facebook.com/cocoaheads) or join us in [Telegram](https://t.me/cocoaheads)!
+
+Also we have [Telegram channels](http://ios-channels.ru) for iOS Developers.
```
Comment (smejl) [inline README.md]: iOS-Channels is independent project, sure, it has a relationship with cch, but it's better to reduce attention to this. Please, add cch logo and reduce size of channels logo.

File: README.md [hunk 3]
```diff
@@ -1 +1,20 @@
...
+
+![iOS Channels](/resources/images/telegram_channels.jpg)
+
+# Wiki
+
+Here you can find all the information about contributing to our open-source projects.
```
Comment (smejl) [inline README.md]: Without the

File: README.md [hunk 4]
```diff
@@ -1 +1,20 @@
...
+
+Now we're working on [CocoaHeads Server](https://github.com/cocoaheadsru/server) — server side api for CocoaHeads Russia app. It was built with Swift and backed by Vapor.
```
Comment (smejl) [inline README.md]: We can create section for projects)

File: git.md [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+
+## Branching
+
+**We use [Trello](https://trello.com) for task managements and [this hook](/resporces/scripts/prepare-commit-msg) for git integration with Trello**
+
+How it works?
+
+Each git brach name should match the following pattern `task/TRELLO_TASK_ID/task-description` **(e.g `task/D84aPyrz/create-models`)**. Hook looks for task id and push it to your commit message. You should write task id once in your branch name.
+
+**You need to place hook in `.git/hooks` folder**.
```
Comment (smejl) [inline git.md]: Please, add about chmod 755 for hook.

File: git.md [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+
+## Commit
+
+- Use only English for commit comments
+- Comments must start from verb and describe main changes **(e.g. Fixed GitHub authorization bug, Added User model)**
```
Comment (smejl) [inline git.md:32]: - Comments must start from verb and describe main changes
`e.g. Fixed GitHub authorization bug, Added User model`

File: git.md [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+  - code belonging to another task
+- Make sure that code compiles
+
+All `git commit` go through code review. Reviewers take common responsibility with commit author for code quality.
+
+### Merge
+
+Commit can be merged if:
+
+- For `develop` and below branches: if there are no warnings, except of *TODO* or *FIXME*
```
Comment (smejl) [inline git.md]: Excellent! 👍

File: git.md [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+- For `master` branch: if there are no warning at all
+- Empty commits (whitespaces, code formatting without changes) won't be merged, to revert use `discard hunk`
+
+## Pull Requests
+
+Pull request must include:
+
+- Trello task's name and link
```
Comment (smejl) [inline git.md:54]: Let's add example for pull requests message.)

File: working_process.md [hunk 9]
```diff
@@ -0,0 +1,40 @@
...
+
+**Xcode → Preferences → Text Editing → Indentation**
+
+![Indentations](/resources/images/indentation.png)
+
+## Coding
+
+### iOS/macOS
+
+**Naming**
```
Comment (smejl) [inline working_process.md]: This and all below is not working process in general. However, it's important information. I think we can write out it.

File: README.md [hunk 10]
```diff
@@ -1 +1,20 @@
- -
+# CocoaHeads Russia
+
+We are the biggest community of iOS/macOS Developers in Russia. We organize developers meetups in Moscow and St.Petersburg.
+
+Follow us on [Twitter](https://twitter.com/cocoaheadsru) and [Facebook](https://www.facebook.com/cocoaheads) or join us in [Telegram](https://t.me/cocoaheads)!
```
Comment (azimin) [inline README.md]: Shouldn't we mention VK group and YouTube channel?

File: README.md (hunk 2 above)
Comment (azimin) [reply README.md]: I would suggest to move it to the bottom and reduce attention from “we”, it’s not CH OS team project :(

File: README.md [hunk 11]
```diff
@@ -1 +1,20 @@
...
+# Wiki
+
+Here you can find all the information about contributing to our open-source projects.
+
+Now we're working on [CocoaHeads Server](https://github.com/cocoaheadsru/server) — server side api for CocoaHeads Russia app. It was built with Swift and backed by Vapor.
+
+## First things first
+
+- Read about [Working Process](working_process.md)
+- And how we organize our [working with Git](git.md)
```
Comment (azimin) [inline README.md]: Does “working” from lowercase works?

File: git.md [hunk 12]
```diff
@@ -0,0 +1,50 @@
...
+
+## General
+
+We use [GitFlow](http://nvie.com/posts/a-successful-git-branching-model/) approach in our projects.
+
+![GitFlow](/resources/images/gitflow.png)
+
+## Branching
+
+**We use [Trello](https://trello.com) for task managements and [this hook](/resporces/scripts/prepare-commit-msg) for git integration with Trello**
```
Comment (smejl) [inline git.md]: By the way, our board is public
https://trello.com/b/CIz6BNT1/server-side-swift

File: git.md (hunk 8 above)
Comment (azimin) [reply git.md:54]: What about predefined structure?

File: working_process.md [hunk 13]
```diff
@@ -0,0 +1,40 @@
...
+These guidelines applied to **all** CocoaHeads Russia projects.
+
+## Xcode and Swift Style Guidelines
+
+**We follow [these guidelines](https://github.com/raywenderlich/swift-style-guide) in coding**
+
+* Don't use acronyms and abbreviations
+* Use indentation == 2 in for *Tab width* and *Indent width*
+
+**Xcode → Preferences → Text Editing → Indentation**
```
Comment (azimin) [inline working_process.md]: We can mention that it should be turned on to project automatically, so there shouldn’t be any problem. Unfortunately VAPOR generates Xcode project from scratch each time with values from Xcode settings.

Comment (smejl) [reply working_process.md]: We can change this configuration with our shell script. We'll write it later.

File: README.md (hunk 4 above)
Comment (petertretyakov) [reply README.md]: I'll rename «Wiki» header to «Projects».

File: git.md (hunk 12 above)
Comment (petertretyakov) [reply git.md]: I'll add Trello link to server docs README. This README is about all CH projects.

File: git.md (hunk 6 above)
Comment (petertretyakov) [reply git.md:32]: Lets use `code formatting` only for code. This is about commit comment so I would suggest to use text formatting here.

File: README.md (hunk 1 above)
Comment (petertretyakov) [reply README.md]: Fixed

File: README.md (hunk 2 above)
Comment (petertretyakov) [reply README.md]: Moved iOS channels to the bottom to «Useful Resources» section. CH logo added

File: README.md (hunk 3 above)
Comment (petertretyakov) [reply README.md]: Fixed

File: git.md (hunk 5 above)
Comment (petertretyakov) [reply git.md]: Fixed

File: working_process.md (hunk 9 above)
Comment (petertretyakov) [reply working_process.md]: Agree. Moved it to separate file `code_style.md`. In working process now there are only working process :)

File: README.md (hunk 10 above)
Comment (petertretyakov) [reply README.md]: Fixed

File: README.md (hunk 11 above)
Comment (petertretyakov) [reply README.md]: Changed it completely :)

File: working_process.md (hunk 13 above)
Comment (petertretyakov) [reply working_process.md]: Fixed

File: git.md (hunk 8 above)
Comment (petertretyakov) [reply git.md:54]: Used template from CH app. I'll add example after first PR from server.

File: git.md (hunk 6 above)
Comment (smejl) [reply git.md:32]: You are right. What about quote? 
> Fixed GitHub authorization bug, Added User model

**Review by smejl — APPROVED** [review]

**Review by azimin — APPROVED** [review]

**Pull request merged**